broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartitionHealth.java

```diff
@@ -34,9 +42,13 @@ public void addFailureListener(final FailureListener failureListener) {
     this.failureListener = failureListener;
   }
 
-  void setHealthStatus(final HealthStatus healthStatus) {
-    final var previousStatus = this.healthStatus;
-    this.healthStatus = healthStatus;
+  private void setHealthStatus(final boolean healthy) {
```

Comment:
I think it makes more sense sense to calculate the health status here instead of in every caller and passing it as a parameter. An `updateHealthStatus` method can be called after updating some condition and that knows how to compute the health status. That way the logic isn't spread out.

Reply:
:+1: I have applied your comments.